kademlia/network.py

```diff
@@ -40,15 +40,20 @@ def __init__(self, ksize=20, alpha=3, id=None, storage=None):
         self.protocol = KademliaProtocol(self.node, self.storage, ksize)
         self.refreshLoop = LoopingCall(self.refreshTable).start(3600)
 
-    def listen(self, port):
+    def listen(self, port, interface=None):
```

Comment:
If you use `interface=''` as the default, then line 51 can just be `return reactor.listenUDP(port, self.protocol, interface=interface)` and you won't need any of the other lines.